packages/js/product-editor/src/hooks/use-product-manager/use-product-manager.ts

```diff
@@ -102,6 +108,25 @@ export function useProductManager< T = Product >( postType: string ) {
 		}
 	}
 
+	async function copyToDraft() {
+		try {
+			const data =
+				AUTO_DRAFT_NAME === prevName && name !== prevName
+					? { name }
+					: {};
```

Comment:
We added this to mimic the `Copy to a new draft` behavior from the legacy editor. When `Copy to a new draft` is pressed on a product that hasn't been saved, yet the name field is filled out, the copied product will retain that name.

Reply:
Done in commit [9bde969](https://github.com/woocommerce/woocommerce/pull/46294/commits/9bde9698cb4f71089ccdbc1e42341fdac0851218)